### DISCUSSION TITLE
Removing duplicate lines.

### DISCUSSION THREAD

**Discussion by borifax** [WinMerge/winmerge#519] [Ideas]
Is it possible to add an option to remove duplicate lines from one of the compared files?

**Comment by sdottaka** [comment]
WinMerge doesn't have the ability to remove duplicate rows, but you can do something similar with regular expression substitutions.

![image](https://user-images.githubusercontent.com/98126/103143289-70ca5580-4756-11eb-90ab-cf2b2fde6388.png)

Or, a general text editor has such a function, so please try using it.

For example, in Notepad ++, see the following URL.

https://stackoverflow.com/questions/3958350/removing-duplicate-rows-in-notepad

**Reply by Gitoffthelawn** [reply]
Interesting idea.

Can you explain that regex a little?  What are the Yen indicating?

**Reply by sdottaka** [reply]
> What are the Yen indicating?

Oh. I forgot to change from the Japanese locale to the English locale.
The yen character has the same meaning as a backslash.

Please refer to the following URL for the meaning of regular expressions.

https://stackoverflow.com/a/16293580

**Reply by Gitoffthelawn** [reply]
Thanks.  I was confused because there was the yen and dollar characters together.  I was thinking you were using the Japanese locale, but the presence of the yen and dollar characters together confused me.  I thought the Japanese locale would replace the dollar character with the yen character, but not replace the backslash with yen!

**Comment by sdottaka** [comment]
The Remove Duplicate Lines plugin has been added to WinMerge version 2.16.14.
![image](https://user-images.githubusercontent.com/98126/133929497-4733b796-4a72-4011-9e01-3d394fd22e27.png)
